packages/components/src/mobile/bottom-sheet/bottom-sheet-navigation/navigation-container.native.js

```diff
@@ -57,67 +61,73 @@ const options = {
 	cardStyleInterpolator: fadeConfig,
 };
 
-const ANIMATION_DURATION = 190;
+const HEIGHT_ANIMATION_DURATION = 300;
+const DEFAULT_HEIGHT = 1;
 
 function BottomSheetNavigationContainer( {
 	children,
 	animate,
 	main,
 	theme,
 	style,
+	testID,
 } ) {
 	const Stack = useRef( createStackNavigator() ).current;
-	const context = useContext( BottomSheetNavigationContext );
-	const [ currentHeight, setCurrentHeight ] = useState(
-		context.currentHeight || 1
+	const navigationContext = useContext( BottomSheetNavigationContext );
+	const { maxHeight: sheetMaxHeight, isMaxHeightSet: isSheetMaxHeightSet } =
+		useContext( BottomSheetContext );
```

Comment:
I'm using it to limit the target height when there's a maximum value set in the bottom sheet component. The height won't exceed the maximum as it's controlled with the `maxHeight` style, but the animation will have a hard stop when it tries to go beyond the maximum value.

| Using max height | Not using max height |
|--------|--------|
| <video src=https://github.com/WordPress/gutenberg/assets/14905380/83a936c2-c7ed-4d25-82f7-07f0cd39ed58> | <video src=https://github.com/WordPress/gutenberg/assets/14905380/9052d4f5-da2a-4191-b754-85d1aa507fe0> | 

Note, how in the Cover block settings, the transition when navigating back from a sub sheet is not smooth.

Reply:
> Also, what value does `isMaxHeightSet` provide over merely checking the value of `maxHeight`? I realize some of this logic was already in place.

AFAIK it tells if the `maxHeight` should be applied. Probably the naming is a bit confusing because it sounds as if it's set, however it's used for determining if it should be set.

https://github.com/WordPress/gutenberg/blob/f7dc1fa0a11c0564770abe3bba3c14ccd4a01f9b/packages/components/src/mobile/bottom-sheet/index.native.js#L360-L364

https://github.com/WordPress/gutenberg/blob/f7dc1fa0a11c0564770abe3bba3c14ccd4a01f9b/packages/components/src/mobile/bottom-sheet/index.native.js#L445-L446

Reply:
Thank you for outlining this. Very helpful.